corehq/apps/accounting/invoicing.py

```diff
@@ -723,6 +725,7 @@ def unit_cost(self):
 
 
 class UserLineItemFactory(FeatureLineItemFactory):
+    history_cls = DomainUserHistory
```

Comment:
It looks like `UserLineItemFactory` already had a child in `WebUserLineItemFactory`, so with a third class added, I'm wondering if it's time to think about the class hierarchy. It seems like a smell that `WebUserLineItemFactory` will inherit the `history_cls` attribute, yet will have no use for it. Do you think that's an issue?

Reply:
After discussing yesterday I think this is basically the solution we talked about - removing the `history_cls` as a class attribute and putting the subclass-specific functionality into something that gets called from within `quantity`. https://github.com/dimagi/commcare-hq/pull/34868/commits/c929dde08be8fcacce5e1aa13b54c8cf5e3cb683

Reply:
Changes look good!